server/mediaDetail/mediaDetailDB.js

```diff
@@ -28,13 +28,33 @@ module.exports = {
   putMediaObjInUserWatchHistory: (userId, mediaObj) => {
     return new Promise((resolve, reject) => {
       let filter = {currentId: userId};
-      let update = {watchHistory: [...watchHistory, mediaObj]};
+      let update = {$push: { history: mediaObj}};
       let options = {new: true, upsert: true};
       User.findOneAndUpdate(filter, update, options, (err, data) => {
         if (err) {
           console.log('ERROR in putMediaObjInUserWatchHistory: ', err);
         } else {
-          console.log('SUCCESS updating media: ', data);
+          resolve(data);
+          console.log('SUCCESS updating media');
+        }
+      });
+    });
+  },
+
+  putNewRating: (mediaId, currentRating, currentRatingCount, rating) => {
+    let newRatingCount = currentRatingCount + 1;
+    let newRating = ((currentRating * currentRatingCount) + rating) / newRatingCount;
+    let newRatingRounded = Number(newRating.toFixed(1));
+    return new Promise((resolve, reject) => {
+      let filter = {id: mediaId};
+      let update = {rating: newRatingRounded, ratingCount: newRatingCount};
+      let options = {new: true, upsert: true};
+      Movie.findOneAndUpdate(filter, update, options, (err, data) => {
+        if (err) {
+          console.log('ERROR in putNewRating: ', err);
```

Comment:
I think here you need to reject on err to handle the error properly. reject(err).

Reply:
Oh yes! Thank you! All fixed!!